### ISSUE TITLE
allow-one-hash-in-raw-strings in clippy.toml not respected for needless_raw_string_hashes

### ISSUE THREAD

**Issue by gwy15** [rust-lang/rust-clippy#11481]
### Summary

in https://doc.rust-lang.org/clippy/lint_configuration.html#allow-one-hash-in-raw-strings, there's a configuration which, as the name suggests, should allow using one hash for raw strings. But it is not currently respected for lint `needless_raw_string_hashes`.

### Reproducer

I tried this code:

`./clippy.toml`:
```
allow-one-hash-in-raw-strings = true
```

`./src/main.rs`:
```rust
const S: &str = r#"
\d+
"#;

fn main() {
    println!("s = {S}");
}
```

I expected to see this happen: No lint.

Instead, this happened:
```
warning: unnecessary hashes around raw string literal
 --> src/main.rs:1:17
  |
1 |   const S: &str = r#"
  |  _________________^
2 | | \d+
3 | | "#;
  | |__^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_raw_string_hashes
  = note: `#[warn(clippy::needless_raw_string_hashes)]` on by default
help: try
  |
1 ~ const S: &str = r"
2 + \d+
3 ~ ";
  |
```

### Version

```text
rustc 1.72.0 (5680fa18f 2023-08-23)
binary: rustc
commit-hash: 5680fa18feaa87f3ff04063800aec256c3d4b4be
commit-date: 2023-08-23
host: aarch64-apple-darwin
release: 1.72.0
LLVM version: 16.0.5
```

**Comment by krtab** [comment]
A commit fixing this issue is in review in the following rust-lang/rust PR clippy subtree:
https://github.com/rust-lang/rust/pull/119552
